covid19-etl/etl/vac_tracker.py

```diff
@@ -0,0 +1,184 @@
+    )
+
+
+class VAC_TRACKER(base.BaseETL):
+    def __init__(self, base_url, access_token, s3_bucket):
+        super().__init__(base_url, access_token, s3_bucket)
+        self.clinical_trials = []
+        self.program_name = "open"
+        self.project_code = "VacTracker"
+        self.metadata_helper = MetadataHelper(
+            base_url=self.base_url,
+            program_name=self.program_name,
+            project_code=self.project_code,
+            access_token=access_token,
+        )
+
+    def files_to_submissions(self):
+        """
+        Reads josn files and converts the data to Sheepdog records
+        """
+        url = "https://biorender.com/page-data/covid-vaccine-tracker/page-data.json"
+        self.parse_file(url)
+
+    def parse_file(self, url):
+        """
+        Converts a json file to data we can submit via Sheepdog. Stores the
+        records to submit in `self.location_data` and `self.time_series_data`.
+        Ignores any records that are already in Sheepdog (relies on unique
+        `submitter_id` to check)
+
+        Args:
+            url (str): URL at which the CSV file is available
+        """
+        print("Getting data from {}".format(url))
+        with closing(requests.get(url, stream=True)) as r:
+            data = r.json()
+            try:
+                for treatment in data["result"]["pageContext"]["treatments"]:
+                    node = treatment["node"]
+                    clinical_trial = self.parse_node(node)
+                    self.clinical_trials.append(clinical_trial)
+            except ValueError as e:
+                print(f"ERROR: value error. Detail {e}")
+
+    def parse_node(self, node):
+        """
+        Converts an element of an JSON file to data we can submit via Sheepdog
+
+        Args:
+            node (dict): node data
+
+        Returns:
+            dict:
+                - clinical trial data, in a format ready to be submitted to Sheepdog
+        """
+        clinical_trial = {
+            "projects": [{"code": self.project_code}],
+            "type": "clinical_trials",
+        }
+
+        for key, value in node.items():
+            if key not in MAP_FIELDS:
+                continue
+            gen3_field = MAP_FIELDS.get(key)[0]
+            gen3_field_type = MAP_FIELDS.get(key)[1]
+            if type(value) != gen3_field_type:
+                print(
+                    f"ERROR: The type of {key} does not match with the one in Gen3. Skip it"
+                )
+                continue
+            if key == "fdaApproved":
+                if "FDA-approved" in value:
+                    value = "Yes"
+                elif value == "":
+                    value = "Unknown"
+                elif value in ["N/A", "N//A", "N/A*"]:
+                    value = "NA"
+                elif value not in ["Yes", "No", "Unknown", "NA", None]:
+                    value = "Unknown"
```

Comment:
there is no mapping for "No"

Reply:
https://github.com/uc-cdis/covid19_datadictionary/blob/996116de759878631bc6d02f7f49c76e83875b3c/gdcdictionary/schemas/clinical_trials.yaml#L144

Reply:
oh i see, the value is already No - ignore my comment :p